### PR TITLE
add optional support for heinleins spamassassin rules

### PR DESCRIPTION
**Pull request by CrawX**
Hi @a16bitsysop,

thanks for providing this pre-configured and multiplatform-enabled rspamd image!

This PR adds support for additional spamassassin rules provided by heinlein-support.de. They host the popular provider mailbox.org and their rules are mainly targeted towards German spam. Some international rules are also included though.
I added bind-tools to use dig to query for the most recent version and a script that downloads & extracts the rules and writes them to a location configured in rspamd. Rspamd is restarted via SIGHUP if the rules changed.

The whole thing is optional and only enabled if HLSLEEP is set.

Thanks!

### CODE REVIEW

File: update_sa_heinlein.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+echo "Updating heinlein rules"
+# Hash current SA rules
+if [[ ! -f /etc/rspamd/custom/sa-rules ]]; then
+  HASH_SA_RULES=0
+else
+  HASH_SA_RULES=$(cat /etc/rspamd/custom/sa-rules | md5sum | cut -d' ' -f1)
+fi
+
+# Deploy
+url="http://www.spamassassin.heinlein-support.de/$(dig txt 1.4.3.spamassassin.heinlein-support.de +short | tr -d '"').tar.gz"
```
Comment (a16bitsysop) [inline update_sa_heinlein.sh]: if this is changed to:
url="http://www.spamassassin.heinlein-support.de/$(drill 1.4.3.spamassassin.heinlein-support.de txt | grep ^1.4.3.spamassassin.heinlein-support.de | cut -d\\" -f2).tar.gz"

Then it works without bind-tools.

**Comment by a16bitsysop** [comment]
Thanks for the merge request, please see comment about changing to not need bind-tools.

File: entrypoint.sh [hunk 2]
```diff
@@ -80,6 +81,17 @@ fi
...
 
+# update heinlein rules once & start daemon to update them in the background
+if [ -n "$HLSLEEP" ]
+then
+  echo 'ruleset = "/etc/rspamd/custom/sa-rules";' > spamassassin.conf
+  mkdir /etc/rspamd/custom/
+  /usr/local/bin/update_sa_heinlein.sh
+  /usr/local/bin/update_sa_heinlein_daemon.sh &
+else
+  rm spamassassin.conf
```
Comment (a16bitsysop) [inline entrypoint.sh]: rm -f is better as does not fail if file does not exist

File: update_sa_heinlein.sh (hunk 1 above)
Comment (CrawX) [reply update_sa_heinlein.sh]: Thanks, I removed bind-tools and replaced the line with your suggestion.

**Comment by CrawX** [comment]
Thanks for the suggestions. I incorporated both and also fixed a typo in the ```ZZZ="${$HLSLEEP-1}"``` to ```ZZZ="${HLSLEEP-1}"```.

**Comment by a16bitsysop** [comment]
Ok will merge it so super linter can check it, and give it a test before building a docker image

**Comment by CrawX** [comment]
@a16bitsysop thanks!

**Pull request merged**